Hw17 resilent4j

KirillKotikov opened the pull request.

Commit:
Resilent4j
[ + ] Добавил модуль с использованием Resilent4j.

Commit:
Merge branch 'main' of https://github.com/KirillKotikov/2023-03-otus-spring-kotikov into hw17-resilent4j

# Conflicts:
#	README.md
#	util/README.md
#	util/pom.xml

Commit:
Resilent4j
[ * ] Изменил последовательно модулей в pom.

vlad-iv commented on util/resilent4j/src/main/java/ru/kotikov/library/controllers/BookRestController.java:
```diff
@@ -0,0 +1,35 @@
+    private final BookService bookService;
+
+    @GetMapping("/api/book/")
+    @TimeLimiter(name = "getAll", fallbackMethod = "getAllBooksFallBack")
```
Предлагаю реализовать метод getAllBooksFallBack

vlad-iv replied:
Предлагаю добавить обработчик ошибок для исключения TimeoutException
Пример:
@ExceptionHandler({TimeoutException.class})
@ResponseStatus(HttpStatus.REQUEST_TIMEOUT)
public void handleTimeoutException() {
}

KirillKotikov replied:
Добавил :)

vlad-iv commented on util/resilent4j/src/main/java/ru/kotikov/library/controllers/BookRestController.java:
```diff
@@ -0,0 +1,35 @@
+    }
+
+    @GetMapping("/api/book/{id}")
+    @TimeLimiter(name = "getById", fallbackMethod = "getBookWithCommentsByBookIdFallBack")
```
Предлагаю реализовать метод getBookWithCommentsByBookIdFallBack

vlad-iv commented on util/resilent4j/src/main/java/ru/kotikov/library/services/BookServiceImpl.java:
```diff
@@ -0,0 +1,61 @@
+    public BookWithCommentDto getBookWithCommentsByBookId(String id) {
+        sleepRandom();
+        BookDto bookDto = BookDto.toDto(bookRepository.findById(id)
+                .orElseThrow(() -> new DataNotFoundException(
```
Предлагаю добавить обработчик ошибок для DataNotFoundException

Commit:
Docker
[ + ] Добавил реализации методов при ошибках и fallBack.
[ + ] Добавил класс шаблон ответа при ошибке.
[ + ] Добавил кэш для ответа при недоступности бд.
